client/my-sites/checkout/checkout/redirect-payment-box.jsx

```diff
@@ -192,10 +195,31 @@ export class RedirectPaymentBox extends PureComponent {
 				analytics.tracks.recordEvent(
 					'calypso_checkout_with_redirect_' + snakeCase( this.props.paymentType )
 				);
+
+				// The Wechat payment type should only redirect when on mobile as redirect urls
+				// are Wechat Pay mobile application urls: e.g. weixin://wxpay/bizpayurl?pr=RaXzhu4
+				if ( 'wechat' === this.props.paymentType ) {
+					const userAgent = new UserAgent().parse( navigator.userAgent );
+
+					if ( ! userAgent.isMobile ) {
+						this.setSubmitState( {
+							info: translate( 'Generating WeChat Payment code.', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

Reply:
+1, use a translator comment here instead (`comment` instead of `context`)